From 1ff4b064f1aca5dbbad5d084435bde2779a63d67 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:32:05 -0400
Subject: [scsi] hpsa: return -ENOMEM not -1 on kzalloc failure in hpsa_get_device_id

Message-id: <1410186734-19986-75-git-send-email-thenzl@redhat.com>
Patchwork-id: 92097
O-Subject: [RHEL6.6 e-stor PATCH 74/83] hpsa: return -ENOMEM not -1 on kzalloc failure in hpsa_get_device_id
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Reviewed-by: Mike Miller <michael.miller@canonical.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit a84d794d5c12a4c118bcdc4fecc2ca76ed02f643)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 4e7ef85..bbd571b 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -2436,7 +2436,7 @@ static int hpsa_get_device_id(struct ctlr_info *h, unsigned char *scsi3addr,
   buflen = 16;
  buf = kzalloc(64, GFP_KERNEL);
  if (!buf)
-  return -1;
+  return -ENOMEM;
  rc = hpsa_scsi_do_inquiry(h, scsi3addr, VPD_PAGE | 0x83, buf, 64);
  if (rc == 0)
   memcpy(device_id, &buf[8], buflen);
-- 
1.7.1